packages/@aws-cdk/aws-glue/lib/database.ts

```diff
@@ -95,13 +123,21 @@ export class Database extends Resource implements IDatabase {
       name: props.databaseName,
     };
 
-    if (props.locationUri !== undefined) {
+    if (props.locationUri !== undefined && props.targetDatabase !== undefined) {
+      throw new Error('locationUri cannot be specified for resource link databases');
```

Comment:
Better to use the actual conflicting property names.

```suggestion
      throw new Error("'locationUri' cannot be specified with targetDatabase'");
```